Add request! function

jwien001 opened the pull request:
There are some cases where I want to raise an exception on request failure. Instead of writing the `with` block every time or writing a helper function in my own code, it would be convenient to have a bang variation of `request`.

Let me know what you think. If this is accepted, it would be great if you could also cut a release after merging so I could start using the new function. Thanks!

dvic commented on test/finch_test.exs:
```diff
@@ -530,6 +530,30 @@ defmodule FinchTest do
+      start_supervised!({Finch, name: finch_name})
+
+      assert_raise(Exception, fn ->
+        Finch.build(:get, "http://idontexist.wat") |> Finch.request(finch_name)
```
Is this test passing? (you're testing `request/2` here instead of `request!/2`)

jwien001 commented:
I didn't know about Credo, but I've fixed the error and now Credo and tests are all passing locally.

jwien001 commented:
It looks like CI was failing because Github deprecated and removed the Ubuntu 18.04 Actions runner image: https://github.com/actions/runner-images/pull/7388. I've updated the runner finch uses to 20.04, which should unblock CI.

jwien001 commented:
Hello, it looks like CI is passing. Do you have any more feedback or anything else I should do to prepare this PR for merging?

jwien001 commented:
Hey @sneako, thanks for merging this in! Would you mind cutting a new release too?

sneako commented:
just published v0.16.0 :tada:
